src/modules/feature-limit/limit-check.service.ts

```diff
@@ -0,0 +1,101 @@
+import { BadRequestException, Injectable, Logger } from '@nestjs/common';
+import { LimitLabels } from './limits.enum';
+import { User } from '../user/user.domain';
+import { SequelizeFeatureLimitsRepository } from './feature-limit.repository';
+import { SequelizeSharingRepository } from '../sharing/sharing.repository';
+import { SharingType } from '../sharing/sharing.domain';
+import { Limit } from './limit.domain';
+import {
+  LimitTypeMapping,
+  MaxInviteesPerItemAttribute,
+} from './limits.attributes';
+
+@Injectable()
+export class LimitCheckService {
+  constructor(
+    private readonly limitsRepository: SequelizeFeatureLimitsRepository,
+    private readonly sharingRepository: SequelizeSharingRepository,
+  ) {}
+
+  private checkFunctions: {
+    [K in LimitLabels]: (params: {
+      limit: Limit;
+      data: LimitTypeMapping[K];
+      user: User;
+    }) => Promise<boolean>;
+  } = {
+    [LimitLabels.MaxSharedItems]: this.isMaxSharedItemsLimitExceeded.bind(this),
+    [LimitLabels.MaxSharedItemInvites]:
+      this.isMaxInviteesPerItemsLimitExceeded.bind(this),
+  };
+
+  checkLimit<T extends keyof LimitTypeMapping>(
+    user: User,
+    limit: Limit,
+    data: LimitTypeMapping[T],
+  ) {
+    const checkFunction = this.checkFunctions[limit.label as LimitLabels];
+    if (!checkFunction) {
+      new Logger().error(
+        `Check function not defined for label: ${limit.label}.`,
+      );
+      return false;
+    }
+    return checkFunction({ limit, data, user });
+  }
+
+  async isMaxSharedItemsLimitExceeded({
+    limit,
+    user,
+  }: {
+    limit: Limit;
+    user: User;
+  }) {
+    const sharingsNumber =
+      await this.sharingRepository.getSharedItemsNumberByUser(user.uuid);
+    const limitExceeded = sharingsNumber >= limit.value;
+    if (limitExceeded) {
+      throw new BadRequestException('You reached the limit of shared items');
+    }
+    return false;
+  }
+
+  async isMaxInviteesPerItemsLimitExceeded({
+    limit,
+    data,
+  }: {
+    limit: Limit;
+    data: MaxInviteesPerItemAttribute;
+  }) {
+    const { itemId, itemType } = data;
```

Comment:
We could add a checker for data here too. If something is missing, just log and throw with a meaningful message.

Reply:
I'm also aware that we need to include the owner here. So we should add 1 extra to the count.